packages/paste-core/components/sidebar/src/navigation/SidebarNavigationDisclosureHeading.tsx

```diff
@@ -52,8 +69,16 @@ const StyledDisclosureHeading = React.forwardRef<HTMLDivElement, SidebarNavigati
           <ChevronDisclosureIcon color="inherit" decorative size="sizeIcon20" />
         </Box>
         {icon ? icon : null}
-        <Box as="span" display="block" marginLeft="space20">
-          {children}
+        <Box
+          as="span"
+          display="block"
+          marginLeft="space20"
+          transition="all 120ms ease"
+          float={visible ? 'none' : 'left'}
```

Comment:
Note: since the wrapping Box uses column-gap and this Box isn't actually hidden, when in collapsed state we see too much paddingRight (due to columnGap). Setting float causes this element to be removed from the flex column-gap property while it is hidden